extensions/typescript-language-features/src/utils/previewer.ts

```diff
@@ -39,24 +39,16 @@ function getTagBodyText(
 		return undefined;
 	}
 
-	// Convert to markdown code block if it is not already one
+	// Convert to markdown code block if it does not already contain one
 	function makeCodeblock(text: string): string {
-		if (text.match(/^\s*[~`]{3}/g)) {
+		if (text.match(/^\s*[~`]{3}/m)) {
```

Comment:
Handling the general case here makes the special casing for `<caption>` redundant